presto-tpch/src/main/java/com/facebook/presto/tpch/statistics/TableStatisticsDataRepository.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.tpch.statistics;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.airlift.tpch.TpchColumn;
+import io.airlift.tpch.TpchTable;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URL;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Optional;
+
+import static com.facebook.presto.util.Optionals.withBoth;
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.lang.String.format;
+
+public class TableStatisticsDataRepository
+{
+    private final ObjectMapper objectMapper;
+
+    public TableStatisticsDataRepository(ObjectMapper objectMapper)
+    {
+        this.objectMapper = objectMapper;
+    }
+
+    public void save(
+            String schemaName,
+            TpchTable<?> table,
+            Optional<TpchColumn<?>> partitionColumn,
+            Optional<String> partitionValue,
+            TableStatisticsData statisticsData)
+    {
+        String filename = tableStatisticsDataFilename(table, partitionColumn, partitionValue);
+        Path path = Paths.get("presto-tpch", "src", "main", "resources", "tpch", "statistics", schemaName, filename + ".json");
+        writeStatistics(path, statisticsData);
+    }
+
+    private void writeStatistics(Path path, TableStatisticsData tableStatisticsData)
+    {
+        File file = path.toFile();
+        file.getParentFile().mkdirs();
+        try {
+            objectMapper
+                    .writerWithDefaultPrettyPrinter()
+                    .writeValue(file, tableStatisticsData);
+        }
+        catch (IOException e) {
+            throw new RuntimeException("Could not save table statistics data", e);
+        }
+    }
+
+    public TableStatisticsData load(String schemaName, TpchTable<?> table, Optional<TpchColumn<?>> partitionColumn, Optional<String> partitionValue)
+    {
+        String filename = tableStatisticsDataFilename(table, partitionColumn, partitionValue);
+        String resourcePath = "/tpch/statistics/" + schemaName + "/" + filename + ".json";
```

Comment:
Path is meant for files and has system-specific part delimiters, see the table here: https://docs.oracle.com/javase/tutorial/essential/io/pathOps.html. In this line we're creating a resourcePath, a path on classpath, that always uses '/' delimiters.